core-bundle/src/Resources/contao/library/Contao/Widget.php

```diff
@@ -230,6 +230,11 @@ public function __set($strKey, $varValue)
 				break;
 
 			case 'value':
+				if (\is_float($varValue))
+				{
+					$varValue = StringUtil::numberToString($varValue);
+				}
+
 				$this->varValue = StringUtil::deserialize($varValue);
```

Comment:
Can we not put this in an `else` condition? Because if the `if` condition matches, `$varValue` can no longer be a serialized string.

Reply:
What if the serialized array contains a collection of floats? Wouldn't we want to convert them to strings then too? 🤔